File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int n = colors.size();
        int res = 0;
        for (int i = 0; i < n; i++) {
            char curr = colors[i];
            int maxVal = INT_MIN;
            int sum = 0;
            while (i < n && colors[i] == curr) {
                sum += neededTime[i];
                maxVal = max(maxVal, neededTime[i]);
                i++;
            }
            res += sum - maxVal; // exclude the largest cost
            if (i < n) {
                i--;
            }
        }
        return res;
    }
};

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        int n = colors.length(), result = 0;
        for (int i = 0; i < n; i++) {
            char curr = colors.charAt(i);
            int maxVal = Integer.MIN_VALUE;
            int sum = 0;
            while (i < n && colors.charAt(i) == curr) {
                sum += neededTime[i];
                maxVal = Math.max(maxVal, neededTime[i]);
                i++;
            }
            result += sum - maxVal; // exclude the largest cost
            if (i < n) {
                i--;
            }
        }
        return result;
    }
}